endaq/ide/nmea.py

```diff
@@ -0,0 +1,179 @@
+                nmea_channel_id = chid
+                break
+    else:  # given NMEA channel
+        nmea_channel_id = channel
+    if nmea_channel_id is None:  # check we found something
+        raise ValueError("No raw NMEA channel specified or found")
+
+    for el in dataset.ebmldoc:
+        if el.name == "ChannelDataBlock":
+            if el[0].value == nmea_channel_id:
+                rawBuffer.extend(el.dump()["ChannelDataPayload"])
+                extractData, rawBuffer = get_nmea(rawBuffer, raw)
+                nmeaSentences.extend(extractData)
+    return nmeaSentences
+
+
+def get_nmea_measurement(data: [NMEAMessage], measure_type, filter_level: int = 0, timestamp="GPS") -> pd.DataFrame:
+    """ Returned dataframe format should appear like dataframes for other channels run through the
+        endaq.ide.to_pandas function.
+
+        :param data: A dataset or list of NMEAMessage Objects. The data to pull measurements from.
+        :param measure_type: Determines the kind of measurements that will be returned in the dataframe. Can accept
+            ANY (returns all following measurements), DIRECTION (degrees, True North), LOCATION (lat/long),
+            and SPEED (km/h)
+        :param filter_level: reliability filter level - Will only process data when connected to this number of
+            satellites. Defaults to 0, maximum of 12.
+        :param timestamp: Default is to provide timestamp from the GPS message. Selecting between GPS time and device
+            time will be the last piece of the job.
+        :raise ValueError: if data is not a list or does not contain NMEAMessage objects
+        :see: NMEA reference <https://www.sparkfun.com/datasheets/GPS/NMEA%20Reference%20Manual-Rev2.1-Dec07.pdf>`
+        :see: NMEAMessage Docs <https://github.com/semuconsulting/pynmeagps/blob/master/pynmeagps/nmeamessage.py>
+    """
+    if not isinstance(data, list) or not isinstance(data[0], NMEAMessage):
+        raise ValueError("Data expected as NMEAMessage objects. Try .get_nmea_sentence()")
+    include, _ = measure.split_types(measure_type)
+    fulldates = []  # we always want the timestamps
+    block = []      # holds NMEA messages before processing
+    rows = []       # holds all rows, used in dataframe construction
+    if measure.ANY in include:
+        # indexes for data columns
+        numCol = 5
+        # checks measurement types
+        wantDirs = wantLocs = wantSpds = True
+        dirCol, latCol, lonCol, spdCol, qualCol = range(numCol)
+        colnames = ["direction", "latitude", "longitude", "speed", "quality"]
+    else:
+        numCol = 0
+        wantDirs = wantLocs = wantSpds = False
+        dirCol = latCol = lonCol = spdCol = -1
+        colnames = []
+        if measure.DIRECTION in include:
+            wantDirs = True
+            dirCol = numCol  # in degrees, degree sign character is \u00B0
+            numCol += 1
+            colnames.append("direction")
```

Comment:
You can probably get rid of numCol and use `len(colnames)` instead. Here (and other similar conditions) you could set the 'col' variable to `len(colnames)` before appending.
```python
dirCol = len(colnames)
colanmes.append('direction')
```